pkg/dependencies/factory.go

```diff
@@ -381,8 +382,20 @@ func (f *Factory) WithExecutableBuilder() *Factory {
 	return f
 }
 
+// ProviderOptions contains per provider options.
+type ProviderOptions struct {
```

Comment:
this allows us to add more options in the future without continually breaking the `WithProvider` function signature.